task-5/pkg/index/index.go

```diff
@@ -0,0 +1,204 @@
+package index
+
+import (
+	"bufio"
+	"fmt"
+	"go-gavrila/task-5/pkg/crawler"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const content_path = `./content.csv`
+const index_path = `./index.csv`
+const separator = `;`
+
+var index = make(map[string][]int)
+var content = []crawler.Document{}
+
+func LoadFiles() {
+	load_content(load_file(content_path))
+	load_index(load_file(index_path))
+}
+
+// Loads data from file and converts it to strings slice
+func load_file(name string) []string {
+	fmt.Println("loading", name)
+	file, err := os.Open(name)
+	if err != nil {
+		fmt.Println(err)
+		return []string{}
+	}
+	defer file.Close()
+
+	return read(file)
+
+}
+
+// reads data from provided source
+func read(r io.Reader) []string {
+
+	reader := bufio.NewReader(r)
+	var lines []string
+
+	for {
+		line, err := reader.ReadString('\n')
+		if err != nil {
+			break
+		}
+		lines = append(lines, strings.TrimSuffix(line, "\n"))
+	}
+
+	fmt.Println("imported", len(lines), "items")
+
+	return lines
+}
+
+// writes a sting to the writer
+func write(w io.Writer, s string) error {
+	_, err := w.Write([]byte(s))
+	return err
+}
+
+// Converts slice of strings to an array of crawler.Document
+func load_content(data []string) {
+
+	for _, line := range data {
+
+		arr := strings.Split(line, separator)
+		i, err := strconv.ParseInt(arr[0], 0, 64)
+		if err != nil {
+			fmt.Println(err)
+		} else {
+			content = append(content, crawler.Document{int(i), arr[1], arr[2], arr[3]})
+		}
+
+	}
+
+}
+
+// Converts strings slice to a map
+func load_index(data []string) {
+
+	for _, line := range data {
+		arr := strings.Split(line, separator)
+		for _, i := range strings.Split(arr[1], `,`) {
+			v, err := strconv.ParseInt(i, 0, 64)
+			if err != nil {
+				fmt.Println(err)
+			} else {
+				index[arr[0]] = append(index[arr[0]], int(v))
+			}
+		}
+	}
+
+}
+
+// Adds documents to a collection and maintains indices.
+func Add(links []crawler.Document) {
+	var counter int
+	if len(content) > 0 {
+		counter = content[len(content)-1].ID
+		// Clear the old index from file if we have previous content
+		if err := os.Truncate(index_path, 0); err != nil {
+			fmt.Println(err)
+			return
+		}
+	}
+
+	content_storage, err := os.OpenFile(content_path, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0666)
```

Comment:
The last parameter really drove me bonkers. I'm sure there is a way to make it OS-independent, but I'm giving up for now. By the way, it is an octal notation of an integer. The leading 0 doesn't affect any access, it simply indicates the notation.